### ISSUE TITLE
Unable to update properties in v3.0.0

### ISSUE THREAD

**Issue by lp-disco** [pulumi/pulumi-akamai#107]
### What happened?

I have seen this issue across multiple stacks when updating properties. Most recent error message below:

m2022/09/14 09:08:06 [ERROR] could not read property version: OperationID=5601cd44-6843-497f-b2e9-bb6c7e33651e PAPI=resourcePropertyUpdate property_version=0 property_name=prp_834888 error="fetching property version: struct validation: PropertyVersion: cannot be blank." contract_id=ctr_M-1MEE94H group_id=grp_156959

I have checked in the control center, and the current property version is 1.

After exporting the stack to JSON, I see the properties have `readVersion` set to zero after each update. A similar issue was reported previously: #50 

### Steps to reproduce

To reproduce this bug:
1. Create a Pulumi stack with a property in it
2. Deploy the stack
3. Make an update to the property's rules
4. Deploy the stack again

This is all I did, it should be enough to show the error.

### Expected Behavior

The property to update, as specified.

### Actual Behavior

Error

### Output of `pulumi about`

```
CLI          
Version      3.31.0
Go Version   go1.17.9
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown

Host     
OS       darwin
Version  12.5.1
Arch     x86_64

This project is written in nodejs (/Users/lpopp/.nvm/versions/node/v14.18.2/bin/node v14.18.2)

Current Stack: gcd-akamai-cdn-eu-dev

TYPE                                            URN
pulumi:pulumi:Stack                             urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::pulumi:pulumi:Stack::vdp-gcd-akamai-cdn-gcd-akamai-cdn-eu-dev
pulumi:providers:aws                            urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::pulumi:providers:aws::aws-eu-west-1
pulumi:providers:aws                            urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::pulumi:providers:aws::aws-ap-southeast-1
pulumi:providers:aws                            urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::pulumi:providers:aws::aws-us-east-1
aws:secretsmanager/secret:Secret                urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::aws:secretsmanager/secret:Secret::akamai-dev-gb-sport-key
pulumi:providers:akamai                         urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::pulumi:providers:akamai::dev-akamai-provider
pulumi:providers:random                         urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::pulumi:providers:random::default_4_6_0
akamai:index/edgeHostName:EdgeHostName          urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::akamai:index/edgeHostName:EdgeHostName::dev-eu-live-hn-gb-sport-live-dev.akamaized.net
random:index/randomPassword:RandomPassword      urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::random:index/randomPassword:RandomPassword::dev-gb-sport-key
aws:secretsmanager/secretVersion:SecretVersion  urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::aws:secretsmanager/secretVersion:SecretVersion::akamai-dev-gb-sport-key
akamai:index/cpCode:CpCode                      urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::akamai:index/cpCode:CpCode::dev-eu-live-cp
akamai:index/property:Property                  urn:pulumi:gcd-akamai-cdn-eu-dev::vdp-gcd-akamai-cdn::akamai:index/property:Property::dev-eu-live-prop


Found no pending operations associated with gcd-akamai-cdn-eu-dev

Backend        
Name           blvl184611m
URL            s3://dev-pulumi-states
User           lpopp
Organizations  
```

### Additional context

Is it possible we are missing something in our configuration somehow?

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by stack72** [comment]
Hey @lp-disco 

I am sorry for the issue here - did this work in v2.9.1? We made a fix there and released that - I am wondering if this is a regression OR if it's still been an issue

Paul

**Comment by lp-disco** [comment]
I actually am not sure. I saw there was a fix for a similar issue, that seemed to be related to `latestVersion`. I believe in my case, `readVersion` is being set to zero after each deployment

**Comment by Hanskun** [comment]
I’m having the same issue when creating properties on version 3.1.1 of @pulumi/akamai. I've attempted to downgrade to 2.9.1 and can successfully update the property/rules on this version. Seems to me the fix in 2.9.1 has been lost on the more recent versions.

**Comment by Hanskun** [comment]
Hi @stack72 ,

Do you know if anyone is involved with this issue?  It's currently quite cumbersome to use the library. 

Since updating properties only work with version 2.9.1, while AppSecActivations are not working with that version, I now have to alias in the newest version for AppSec configuration changes, while using the old version for managing the Akamai property. Would be great if someone was able to resolve the issue :)

**Comment by guineveresaenger** [comment]
Hi @Hanskun - `stack72` no longer is a maintainer for this repository so we apologize for having lost a bit of context here.

We're currently [updating to a new major version](https://github.com/pulumi/pulumi-akamai/pull/121) and would love to hear back if this issue is still a problem.

**Comment by Hanskun** [comment]
Hi @guineveresaenger 

It was still a problem on the current 3.1.1 release, but I have not attempted to verify it towards the new major version. I could try to see if it helps, but it is not currently available as a release is it? As it is now I have to use version 2.9.1 to do an update for a property in Node.

**Comment by guineveresaenger** [comment]
Hi @Hanskun - apologies for the holiday break.

I just cut pulumi-akamai [4.0.0](https://github.com/pulumi/pulumi-akamai/releases/tag/v4.0.0) and would love to hear back.

**Comment by Hanskun** [comment]
Hi @guineveresaenger ,

I've finally attempted to upgrade to the 4.0.0 version, but I am still seing the same issue in this version. I am seeing the same issue as before

pulumi:pulumi:Stack <stackname> running 2023/01/17 10:00:52 [ERROR] could not read property version: OperationID=36dc529e-5ca5-4bd4-9e6c-d5f84abd4e4d PAPI=resourcePropertyUpdate property_name=<propertyname> contract_id=<contractid> group_id=<groupid> property_version=0 error="fetching property version: struct validation: PropertyVersion: cannot be blank."

This issue is not seen in version 2.9.1, which I think had a [fix](https://github.com/pulumi/pulumi-akamai/issues/101) for this. For me that problem resurfaced in the following releases.

**Comment by jjgrinwis** [comment]
akamai:index:Property (john.pulumi.nl):
    error: 1 error occurred:
        * updating urn:pulumi:betajam::property::akamai:index/property:Property::john.pulumi.nl: 1 error occurred:
        * fetching property version: struct validation: PropertyVersion: cannot be blank.

**Comment by guineveresaenger** [comment]
Thank you for verifying for us. We'll take a look as soon as we can.

**Comment by iwahbe** [comment]
It looks like this was fixed previously in #50 and #101. The fix for #101 looked like https://github.com/pulumi/terraform-provider-akamai/commit/db62822ff12d95f5a1ada08a7cdaec33a177de8c, and was released as part of the upgrade to upstream v2.2.0.

**Comment by iwahbe** [comment]
I was able to replicate with the following program:

```typescript
let groupId = "grp_201328";
let productId = "prd_Fresca";
const contractId = akamai
  .getContract({
    groupId: groupId,
  })
  .then((x) => x.id);

const prop = new akamai.Property("test", {
  productId: productId,
  contractId: contractId,
  groupId: groupId,
  hostnames: initialCreate ? [] : [
    {
      cnameFrom: "example.com",
      cnameTo: "example.com.edgekey.net",
      certProvisioningType: "DEFAULT",
    },
  ],
  ruleFormat: "v2020-03-04",
});
```

I set `initialCreate` to true for the initial creation, then set it to false on the next `pulumi up` to induce an update.
